### PR TITLE
Replace HslColor for HSLColor from painting

### PR DESCRIPTION
**Pull request by calvintam236**

### CODE REVIEW

**Comment by calvintam236** [comment]
@WieFel any thoughts?

**Comment by WieFel** [comment]
@calvintam236 sorry, I forgot to press submit review button 😬

File: lib/src/tinycolor.dart [hunk 1]
```diff
@@ -104,9 +95,8 @@ class TinyColor {
 
   TinyColor darken([int amount = 10]) {
     final hsl = this.toHsl();
-    hsl.l -= amount / 100;
-    hsl.l = clamp01(hsl.l);
-    return TinyColor.fromHSL(hsl);
+    final lightness = clamp01(hsl.lightness - amount / 100);
+    return TinyColor.fromHSL(hsl.withLightness(lightness));
```
[the remark this answers is not in the document]
Comment (dsyrstad) [reply lib/src/tinycolor.dart:97]: Having unit tests as "vises" before the refactoring would ensure that it doesn't break anything.

[the remark this answers is not in the document]
Comment (calvintam236) [reply lib/src/tinycolor.dart:97]: Unit test is a good idea. If you can write them, it would be great.

**Comment by calvintam236** [comment]
@WieFel @dsyrstad do you have a list of expected result from test? Upon checking, the test is basically empty, not checking for output. I can write the test if someone can gave me the parameter input and expected output.

**Comment by WieFel** [comment]
I would say
1. Copy every "old" version of the functions I commented on.
2. Put them in the test file and name them e.g. `lightenOld()`, etc.
3. For every function, loop from through the range of min to max allowed value. E.g. in the case of lighten from 0 to 100, for example with step of 10.
4. Compare output of "old" with "new" function and assert they are equal.

**Comment by calvintam236** [comment]
> I would say
> 
> 1. Copy every "old" version of the functions I commented on.
> 2. Put them in the test file and name them e.g. `lightenOld()`, etc.
> 3. For every function, loop from through the range of min to max allowed value. E.g. in the case of lighten from 0 to 100, for example with step of 10.
> 4. Compare output of "old" with "new" function and assert they are equal.

I am referring to writing tests for CI. It does not work for testing in the future.

**Comment by WieFel** [comment]
@calvintam236 sorry for not answering. I am very busy with work.
Just try to come up with some ideas yourself and we can discuss it ;)

**Comment by calvintam236** [comment]
Blocked by #35

**Comment by calvintam236** [comment]
@WieFel I just added test for HSL. They are passed so it should be okay.

**Comment by calvintam236** [comment]
Note that the `ci/format` test can be ignored at this moment as the tests will be updated again in #41.

**Review by WieFel — APPROVED** [review]

**Pull request merged**